tests/integration/test_stationboard.py

```diff
@@ -1,47 +1,130 @@
-
-    def test_stationboard_with_cb(self):
-        params = {'callback': 'cb_'}
-        resp = self.testapp.get('/stationboard/stops/8501120', params=params, status=200)
-        self.assertEqual(resp.content_type, 'application/javascript')
-        resp.mustcontain('cb_(')
-
-    def test_stationboard_wrong_id(self):
-        resp = self.testapp.get('/stationboard/stops/toto', status=400)
-        resp.mustcontain('The id must be an integer')
-
-    def test_stationboard_wrong_limit(self):
-        params = {'limit': 'toto'}
-        resp = self.testapp.get('/stationboard/stops/8501120', params=params, status=400)
-        resp.mustcontain('The limit parameter must be an integer')
+        self.mock_api_key = "dummy_api_key"
+        self.mock_url = "https://dummy-url.com"
+
+    def generate_mock_response(self, departures, now):
+        """Generate a dynamic mock OJP response with correct structure."""
+        departure_events = "".join(
+            f"""
+            <StopEventResult>
+                <StopEvent>
+                    <ThisCall>
+                        <CallAtStop>
+                            <siri:StopPointRef>{dep['id']}</siri:StopPointRef>
+                            <StopPointName>
+                                <Text>{dep['label']}</Text>
+                            </StopPointName>
+                            <ServiceDeparture>
+                                <TimetabledTime>{dep['departureDate']}</TimetabledTime>
+                                <EstimatedTime>{dep.get('estimatedDate', dep['departureDate'])}</EstimatedTime>
+                            </ServiceDeparture>
+                        </CallAtStop>
+                    </ThisCall>
+                    <Service>
+                        <PublishedServiceName>
+                            <Text>{dep['label']}</Text>
+                        </PublishedServiceName>
+                        <DestinationText>
+                            <Text>{dep['destinationName']}</Text>
+                        </DestinationText>
+                        <DestinationStopPointRef>{dep['destinationId']}</DestinationStopPointRef>
+                    </Service>
+                </StopEvent>
+            </StopEventResult>
+            """
+            for dep in departures
+        )
+        return f"""<?xml version="1.0" ?>
+        <OJP xmlns:siri="http://www.siri.org.uk/siri" xmlns="http://www.vdv.de/ojp" version="2.0">
+            <OJPResponse>
+                <siri:ServiceDelivery>
+                    <siri:ResponseTimestamp>{now}</siri:ResponseTimestamp>
+                    <OJPStopEventDelivery>
+                        <siri:ResponseTimestamp>{now}</siri:ResponseTimestamp>
+                        {departure_events}
+                    </OJPStopEventDelivery>
+                </siri:ServiceDelivery>
+            </OJPResponse>
+        </OJP>"""
+
+    @patch.object(opentransapi.OpenTrans, "send_post")
```

Comment:
Why mocking your library OpenTrans.send_post ? you don't fully test it this way. You should mock the requests library, you can use https://requests-mock.readthedocs.io/en/latest/overview.html to do so, this will can the post and return your mocked response and you can fully test your code.

Reply:
Also the opentrans API key should be removed from CI see https://github.com/geoadmin/infra-terraform-bgdi/blob/dd382ba8c135d491f8b55803925baba501973910/tf/aws/swisstopo-bgdi-builder/codebuild_projects/mf-chsdi3/buildspec.yml#L23C5-L23C22

Reply:
Is the key in SSM already the new one ? https://github.com/geoadmin/infra-terraform-bgdi/blob/dd382ba8c135d491f8b55803925baba501973910/tf/aws/swisstopo-bgdi-builder/shared/secrets/opentrans_api.tf#L1

Reply:
> Is the key in SSM already the new one ? https://github.com/geoadmin/infra-terraform-bgdi/blob/dd382ba8c135d491f8b55803925baba501973910/tf/aws/swisstopo-bgdi-builder/shared/secrets/opentrans_api.tf#L1

If it comes from this PR, then yes: https://github.com/geoadmin/infra-terraform-bgdi/pull/509

Reply:
> Also the opentrans API key should be removed from CI see https://github.com/geoadmin/infra-terraform-bgdi/blob/dd382ba8c135d491f8b55803925baba501973910/tf/aws/swisstopo-bgdi-builder/codebuild_projects/mf-chsdi3/buildspec.yml#L23C5-L23C22. And in order to make sure that the mocking is working, the key MUST BE removed before merging this PR so we can verify it. Would be nice to link the key removal PR to this PR.

Guess removal of the old secrets from CI needs to be done all together with the deployment of the new release of mf-chsdi3. Otherwise the CI will be broken for some time, which would not be a nice state.